File: python/task1/S1_VAR_07.py
# S1_VAR_07 - NaN and math.isnan

import math

nan1 = float("nan")

try:
    nan2 = 0.0 / 0.0
except ZeroDivisionError:
    nan2 = float("nan")

print(f"nan1 = float('nan'): {nan1}")
print(f"nan2 (from 0.0/0.0 or fallback): {nan2}")
print()
print(f"nan1 == nan1: {nan1 == nan1}")  # False! NaN != NaN
print(f"nan1 == nan2: {nan1 == nan2}")  # False
print()
print(f"math.isnan(nan1): {math.isnan(nan1)}")
print(f"math.isnan(nan2): {math.isnan(nan2)}")
print(f"math.isnan(42): {math.isnan(42)}")

File: js/task1/S1_VAR_07.js
// S1_VAR_07 - Coercion via unary +
// Unary + converts strings to numbers, returns NaN for invalid strings

function toNumberOrNull(x) {
  if (typeof x !== "string") return null;
  const num = +x;
  if (Number.isNaN(num)) return null;
  return num;
}

// Test cases
const testValues = [
  { label: '"12"', value: "12" },
  { label: '"12.5"', value: "12.5" },
  { label: '" 12 "', value: " 12 " },
  { label: '"12x"', value: "12x" },
  { label: '""', value: "" }
];

console.log("Input".padEnd(12) + "toNumberOrNull()");
console.log("-".repeat(30));

for (const t of testValues) {
  const result = toNumberOrNull(t.value);
  console.log(t.label.padEnd(12) + (result === null ? "null" : result));
}

// Note on empty string: +"" returns 0, which is technically a valid number
// The exercise doesn't specify handling this edge case, so we return 0
